### PR TITLE
Adding Code for Sign Up, Sign In and Sign Out endpoints.

### PR DESCRIPTION
**Pull request by prateek4sep**
Please review.

### CODE REVIEW

File: quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        userEntity.setFirstName(signupUserRequest.getFirstName());
+        userEntity.setLastName(signupUserRequest.getLastName());
+        userEntity.setUserName(signupUserRequest.getUserName());
+        userEntity.setEmail(signupUserRequest.getEmailAddress());
+        userEntity.setPassword(signupUserRequest.getPassword());
+        userEntity.setCountry(signupUserRequest.getCountry());
+        userEntity.setAboutMe(signupUserRequest.getAboutMe());
+        userEntity.setDob(signupUserRequest.getDob());
+        userEntity.setRole("nonadmin");
+        userEntity.setContactNumber(signupUserRequest.getContactNumber());
```
Comment (saidarshan27) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java]: I think we can create a private utility method which will take signupUserRequest and return us a userEntity, I guess that would be more cleaner

**Review by saidarshan27 — COMMENTED** [review]
eveerything seems good to me prateek, one suggestion is about creating the userEntity in the signup controller method.

File: quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java (hunk 1 above)
Comment (prateek4sep) [reply quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java]: Done. Added a separate method.

**Review by bhushansharma — APPROVED** [review]
Everything looks good

**Review by ShubhamDhoot — APPROVED** [review]
All looks good.

**Pull request merged**